p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/repository/DefaultRepositoryIdManager.java

```diff
@@ -52,6 +57,15 @@ public void addMapping(String mavenRepositoryId, URI location) {
             logger.warn("p2 repository with URL " + key + " is associated with multiple IDs; was '" + previousValue
                     + "', now is '" + mavenRepositoryId + "'");
         }
+
+		// Checks if a mirror exists for the given location
+		MavenRepositoryLocation repositoryLocation = new MavenRepositoryLocation(mavenRepositoryId, key);
+		MavenRepositoryLocation mirrorLocation = settings.getMirror(repositoryLocation);
+		if (mirrorLocation != null) {
+			// A mirror exists. Adds its id to the map of known mirror ids.
+			URI mirrorKey = normalize(mirrorLocation.getURL());
+			knownMavenMirrorIds.put(mirrorKey, mirrorLocation.getId());
```

Comment:
instead of having an extra map, can we probably just add it to the existing `knownMavenRepositoryIds` map?
Also this should include the above check about duplicate mapped ids.

Reply:
Sure, I can use just one map. I implemented it this way because, initially, I was tempted to propose modifying the _IRepositoryIdManager_ interface so that repositories were not mixed with mirrors in the _getKnownMavenRepositoryLocations_ method.

Regarding duplicate ids when dealing with mirrors, the same mirror may be used for several repository locations and that is something I don't think there is any need to warn the user about.

Reply:
Thinking a bit more about that I wonder if we need to store the mirror IDs at all... the id map is actually there because in some cases we don't know the id of the server, but for a _mirror_ we can simply fetch all mirrors from the current maven session like we do in `DefaultMavenRepositorySettings.initialize()`.

So I think `MavenRepositorySettings` should get a new method `MavenRepositorySettings#getMirrors()` and this can then be used in `getKnownMavenRepositoryLocations()` to simply return all mirror `id` -> `url` mappings.